test/integration/src/test/java/io/pravega/test/integration/KVTableWithStreamTest.java

```diff
@@ -0,0 +1,769 @@
+        }
+
+        // Creating KVT2 under Scope2.
+        assertTrue("The requested KVT must get created",
+                this.controller.createKeyValueTable(newScope2, newKVT2.getKeyValueTableName(),
+                        config).join());
+
+        assertEquals("Table partition value should be same as passed", config.getPartitionCount(),
+                Futures.getAndHandleExceptions(this.controller.getCurrentSegmentsForKeyValueTable(newScope2,
+                        newKVT2.getKeyValueTableName()), RuntimeException::new).getSegments().size());
+
+        val tableSegments2 = this.controller.getCurrentSegmentsForKeyValueTable(newScope2,
+                newKVT2.getKeyValueTableName()).join();
+
+        for (val s : tableSegments2.getSegments()) {
+            // We know there's nothing in these segments. But if the segments hadn't been created, then this will throw
+            // an exception.
+            this.tableStore.get(s.getKVTScopedName(), Collections.singletonList(new ByteArraySegment(new byte[1])), TIMEOUT).join();
+        }
+
+        // seal the stream.
+        CompletableFuture<Boolean> sealStreamStatus = controller.sealStream(newScope2, streamName2);
+        log.info("Sealing stream {}", streamName2);
+        assertTrue(sealStreamStatus.get());
+        // delete the stream.
+        CompletableFuture<Boolean> deleteStreamStatus = controller.deleteStream(newScope2, streamName2);
+        log.info("Deleting stream {}", streamName2);
+        assertTrue(deleteStreamStatus.get());
+
+        // Stream and its stream segments should get deleted but KVTable and its table segments should +
+        // exist and we should be able to read/write from them.
+        for (val s : strmSegments2.getSegments()) {
+            AssertExtensions.assertSuppliedFutureThrows(
+                    "Segment " + s + " has not been deleted.",
+                    () -> this.streamSegmentStore.getStreamSegmentInfo(s.getScopedName(), TIMEOUT),
+                    ex -> ex instanceof StreamSegmentNotExistsException);
+        }
+
+        for (val s : tableSegments2.getSegments()) {
+            // We know there's nothing in these segments. But if the segments hadn't been created, then this will throw
+            // an exception.
+            this.tableStore.get(s.getKVTScopedName(), Collections.singletonList(new ByteArraySegment(new byte[1])), TIMEOUT).join();
+        }
+
+        KeyValueTableFactory newKeyValueTableFactory = new KeyValueTableFactoryImpl(newScope2, this.controller, this.connectionPool);
+        KeyValueTable<Integer, String> newKeyValueTable = newKeyValueTableFactory.forKeyValueTable(newKVT2.getKeyValueTableName(),
+                KEY_SERIALIZER, VALUE_SERIALIZER, KeyValueTableClientConfiguration.builder().build());
+
+        String keyFamily = "KeyFamily1";
+        LinkedHashMap<Integer, String> entryMap1 = new LinkedHashMap<>();
+        Integer[] keyArray = new Integer[10];
```

Comment:
This test has a lot of code that is identical to the test above. Can you factor it out and reuse it?

The same comment as above holds: merge these 3 loops in 1.

Reply:
Sure, I'll refactor the code here.